packages/@react-stately/color/src/Color.ts

```diff
@@ -128,6 +138,82 @@ class RGBColor extends Color {
     return this.red << 16 | this.green << 8 | this.blue;
   }
 
+  private toHSB(): Color {
+    let r = this.red / 255;
+    let g = this.green / 255;
+    let b = this.blue / 255;
+    let max = Math.max(r, g, b);
+    let min = Math.min(r, g, b);
+    let h: number;
+    let s: number;
+    let d = max - min;
+
+    s = max === 0 ? 0 : d / max;
+
+    if (d === 0) {
+      h = 0; // achromatic
+    } else {
+      switch (max) {
+        case r:
+          h = (g - b) / d + (g < b ? 6 : 0);
+          break;
+        case g:
+          h = (b - r) / d + 2;
+          break;
+        case b:
+          h = (r - g) / d + 4;
+          break;
+      }
+
+      h /= 6;
+    }
+
+    return new HSBColor(
+      h * 360,
+      s * 100,
+      b * 100,
+      this.alpha
+    );
+  }
+
+  private toHSL(): Color {
+    let r = this.red / 255;
+    let g = this.green / 255;
+    let b = this.blue / 255;
+    let max = Math.max(r, g, b);
+    let min = Math.min(r, g, b);
+    let h: number;
+    let s: number;
+    let l = (max + min) / 2;
+
+    if (max === min) {
+      h = s = 0; // achromatic
+    } else {
+      let d = max - min;
+      s = d / (l < .5 ? max + min : 2 - max - min);
+
+      switch (max) {
+        case r:
+          h = (g - b) / d + (g < b ? 6 : 0);
+          break;
+        case g:
+          h = (b - r) / d + 2;
+          break;
+        case b:
+          h = (r - g) / d + 4;
+          break;
+      }
+
+      h /= 6;
+    }
+
+    return new HSLColor(
+      h * 360,
+      s * 100,
+      l * 100,
+      this.alpha);
+  }
```

Comment:
For clarity:

```ts
  /**
   * Converts an RGB color value to HSL.
   * Conversion formula adapted from https://en.wikipedia.org/wiki/HSL_and_HSV#From_RGB.
   * @returns An HSLColor object.
   */
  private toHSL(): Color {
    const red = this.red / 255;
    const green = this.green / 255;
    const blue = this.blue / 255;
    const min = Math.min(red, green, blue);
    const max = Math.max(red, green, blue);
    const lightness = (max + min) / 2;
    const chroma = max - min;
    let hue: number;
    let saturation: number;
    
    if (chroma === 0) {
      hue = saturation = 0; // achromatic
    } else {
      saturation = chroma / (lightness < .5 ? max + min : 2 - max - min);

      switch (max) {
        case red:
          hue = (green - blue) / chroma + (green < blue ? 6 : 0);
          break;
        case green:
          hue = (blue - red) / chroma + 2;
          break;
        case blue:
          hue = (red - green) / chroma + 4;
          break;
      }

      hue /= 6;
    }

    return new HSLColor(
      hue * 360,
      saturation * 100,
      lightness * 100,
      this.alpha);
  }
```

Reply:
fromRGB HSL and HSV:


[<img width="523" alt="Screen Shot of conversion function converting from RGB to HSL or HSV on wikipedia" src="https://user-images.githubusercontent.com/154077/139709533-d24b4364-1b66-40fe-b98e-d8f98c612357.png">](https://en.wikipedia.org/wiki/HSL_and_HSV#From_RGB)